library/core/src/slice/sort.rs

```diff
@@ -365,12 +365,12 @@ where
         if count > 0 {
             macro_rules! left {
                 () => {
-                    l.offset(*start_l as isize)
+                    l.add(*start_l as usize)
                 };
             }
             macro_rules! right {
                 () => {
-                    r.offset(-(*start_r as isize) - 1)
+                    r.sub((*start_r as usize) + 1)
```

Comment:
`*start_r` is `usize`; `-(-a - b) = a + b`

Reply:
Hmm, `start_r = MaybeUninit::slice_as_mut_ptr(&mut offsets_r);` (line 337), so isn't `*start_r` a `u8`?

(And being `u8` means the `+1` conveniently is definitely not going to overflow.)

Reply:
Oh, right. If it was `usize` I wouldn't left the cast here 🤦🏻